src/firmware/app/primitives/estop_primitive.c

```diff
@@ -0,0 +1,25 @@
+#include "firmware/app/primitives/estop_primitive.h"
+
+typedef struct EstopPrimitiveState
+{
+} EstopPrimitiveState_t;
+DEFINE_PRIMITIVE_STATE_CREATE_AND_DESTROY_FUNCTIONS(EstopPrimitiveState_t)
+
+void app_estop_primitive_start(TbotsProto_EstopPrimitive prim_msg, void* void_state_ptr,
+                               FirmwareWorld_t* world)
+{
+    app_primitive_makeRobotSafe(world);
+}
+
+static void app_estop_primitive_tick(void* void_state_ptr, FirmwareWorld_t* world)
+{
+    app_primitive_makeRobotSafe(world);
+}
+
+/**
+ * \brief The estop movement primitive.
+ */
+const primitive_t ESTOP_PRIMITIVE = {.direct        = true,
```

Comment:
Checking if the value of `direct` matters here? I know we want to remove it, but does the estop primitive need to be direct to work with the old firmware? IIRC `direct` changed behaviour with bypassing the e-stop (sometimes), so want to make sure this won't cause weird issues.

Reply:
I was interpreting direct to mean direct control over the capacitors, so in the primitive manager, I've added
```
    if (!manager->current_primitive->direct)
    {
        // Charge capacitor during gameplay
        const FirmwareRobot_t *robot = app_firmware_world_getRobot(world);
        Charger_t *charger           = app_firmware_robot_getCharger(robot);
        app_charger_charge_capacitor(charger);
    }
```
which means that for primitive that are not direct, we automatically charge the capacitor.

Reply:
that makes sense. You might just want to double check the code in `io/receive.c` in the old firmware, since IIRC it used the value of `direct` for some conditional logic before entering the primitive manager.

Reply:
hm, so receive.c doesn't use this, but mrf does, but I think this is a direct bit in the message rather than in the primitive

Reply:
Oh there's a different `direct` for the message vs the primitive? I don't remember that. If you've checked the behaviour is ok with these changes I'm fine, that just seems really odd to me (that's there's 2 directs).

Reply:
OK I dug around for the real history
1. Originally, in https://github.com/UBC-Thunderbots/Software/pull/487, direct was ported over as a field in primitive_t and receive.c would check primitive_is_direct to do primitive logic. This would be checking a bit in a received `word` that would represent the primitive (I poked around https://github.com/UBC-Thunderbots/Software/commit/7782bf06 if you want a look)
2. By way of https://github.com/UBC-Thunderbots/Software/pull/1145, https://github.com/UBC-Thunderbots/Software/pull/1580, and others, direct is now a field in the primitive proto and is used in primitive manager rather than in receive.c. receive.c just passes the primitive over to primitive manager doesn't deal with the direct logic at all

I don't think any duplication exist since primitive protobufs should only handled in the primitive manager